src/SimpleCollection/AbstractCollection.php

```diff
@@ -344,39 +340,33 @@ public function getKeys()
     }
 
     /**
-     * Filters the current values and return a new collection
-     *
+     * Produces an immutable version of the list and filters for values
      * @param \Closure $cClosure
-     *
      * @return AbstractCollection
      */
     public function filter(\Closure $cClosure)
     {
-        $sClassName      = get_class($this);
-        $aFilteredValues = array();
-        foreach ($this->values as $sKey => $mValue) {
-            if (true === $cClosure($mValue, $sKey)) {
-                $aFilteredValues[$sKey] = $mValue;
+        $modifiedCollection = clone $this;
+        foreach ($modifiedCollection->values as $sKey => $mValue) {
+            if (false === call_user_func($cClosure, $mValue, $sKey)) {
```

Comment:
Why call_user_func? Imho to call the closure dynamicly is faster and use lesser memory.